### PR TITLE
Add team tree, fix TeamLoader bugs

### PR DESCRIPTION
**Pull request by mlsteele**
Adds `keybase team show-tree`. Also fix two `TeamLoader` bugs.

`keybase team show-tree teamname` will show the tree from the root ancestor of teamname.
If you are the admin of the root, it looks like this
```
$ kbu team show-tree jazafraz
- jazafraz
  - jazafraz.aa3
  - jazafraz.aa4
    - jazafraz.aa4.bb1
    - jazafraz.aa4.bb2
```

If you are not the admin of the root, it looks like this:
```
$ kbu team show-tree jazafraz
- jazafraz *
  - jazafraz.aa4 *
    - jazafraz.aa4.bb2
* Subteams of teams you are not an admin of may be hidden.
```
Where that user is a writer of `aa4`, and an admin of `bb2`.

TeamLoader bug 1 was that `me` and `subteamReader` were not plumbed through, so some loads failed.

TeamLoader bug 2 was trickier and has to do with mid-team renaming. I ran in to it while testing `show-tree`. Say there is a parent team X.P1 which spawned a subteam S and then got renamed to X.P2. If you are added to P1 and load it then you will have the `new_subteam` link for S stubbed out. Then if you are added to S and try to inflate that link on the P2 chain, it will fail because "X.P1.S" is not a good name for a subteam of "X.P2".

The fix in this PR is to remember previous names for the subteam. It was getting confusing to remember which parts of team names were verified. So I split up the storage. `TeamSigChainState` now only stores the stuff we know for sure. And `TeamData.Name` stores the latest full name garnered by loading the whole ancestor chain. Hopefully the avdl is clear. I believe if there are no bugs that this should provide the same guarantees for team names except without that bug.

The team cache got invalidated (`diskStorageVersion++`) as a result of this change.

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
The commits eec1cc19007b...38d8dc491dbb introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1500331515923-eec1cc19007b-38d8dc491dbb-linux/Markdown-emoji-eec1cc19007b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1500331515923-eec1cc19007b-38d8dc491dbb-linux/Markdown-emoji-38d8dc491dbb.png)  
   Markdown-emoji comparison
</summary>

File: go/client/cmd_team_show_tree.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	if err != nil {
+		return err
+	}
+
+	protocols := []rpc.Protocol{
+		NewSecretUIProtocol(v.G()),
+	}
+	if err = RegisterProtocolsWithContext(protocols, v.G()); err != nil {
+		return err
+	}
```
Comment (patrickxb) [inline go/client/cmd_team_show_tree.go]: You don't need SecretUI.

File: go/client/cmd_team_show_tree.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	SessionID int
+	libkb.Contextified
+}
+
+func (v *CmdTeamShowTree) ParseArgv(ctx *cli.Context) (err error) {
+	v.TeamName, err = ParseOneTeamNameK1(ctx)
+	if err != nil {
+		return err
+	}
+	return err
```
Comment (patrickxb) [inline go/client/cmd_team_show_tree.go]: just one `return err`

File: go/client/cmd_team_show_tree.go (hunk 1 above)
Comment (mlsteele) [inline go/client/cmd_team_show_tree.go]: I need to decrypt cached team snapshots in the loader. Does that change things?

File: go/teams/rename_test.go [hunk 3]
```diff
@@ -54,3 +54,59 @@ func TestRenameSimple(t *testing.T) {
 	require.Len(t, parent.chain().inner.SubteamLog, 1, "subteam log has 1 series")
 	require.Len(t, parent.chain().inner.SubteamLog[*subteamID], 2, "subteam log has 2 entries")
 }
+
+// This was a bug that once caused the loader to get stuck.
+// - Create team A
+// - Create team A.B1
+// - Create team A.B1.C <new_subteam>
+// - Rename A.B1 -> A.B2 (implicitly renaming A.B1.C -> A.B2.C)
+// - Someone else (U1) loads A.B1 with <new_subteam> stubbed
```
Comment (patrickxb) [inline go/teams/rename_test.go]: How does the load of A.B1 work if it is already renamed to A.B2?

Comment (mlsteele) [inline go/teams/rename_test.go]: I meant they load that team, by ID or the new name. Will change comment.

File: go/client/cmd_team_show_tree.go (hunk 1 above)
Comment (patrickxb) [inline go/client/cmd_team_show_tree.go]: Nope.  And you aren't sending SecretUI anywhere in the service, so this is essentially ignored.

File: go/teams/rename_test.go (hunk 3 above)
Comment (patrickxb) [inline go/teams/rename_test.go]: ok, cool

File: go/client/cmd_team_show_tree.go (hunk 1 above)
Comment (mlsteele) [inline go/client/cmd_team_show_tree.go]: Cool, same with `cmd_team_rename` then too. Do I need `RegisterProtocolsWithContext` at all?

Comment (patrickxb) [inline go/client/cmd_team_show_tree.go]: Nope.

**Pull request merged**